internal/executor/rpc/rpc.go

```diff
@@ -0,0 +1,310 @@
+		return assumedBridgeIP
+	}
+
+	addrs, err := iface.Addrs()
+	if err != nil {
+		return assumedBridgeIP
+	}
+
+	if len(addrs) > 1 {
+		ip, _, err := net.ParseCIDR(addrs[0].String())
+		if err != nil {
+			return assumedBridgeIP
+		}
+
+		return ip.String()
+	}
+
+	return assumedBridgeIP
+}
+
+// Start creates the listener and starts RPC server in a separate goroutine.
+func (r *RPC) Start() {
+	host := "localhost"
+
+	// Work around host.docker.internal missing on Linux
+	//
+	// See the following tickets:
+	// * https://github.com/docker/for-linux/issues/264
+	// * https://github.com/moby/moby/pull/40007
+	if runtime.GOOS == "linux" {
+		host = getDockerBridgeIP()
+	}
+
+	address := fmt.Sprintf("%s:0", host)
+
+	listener, err := net.Listen("tcp", address)
+	if err != nil {
+		panic(err)
+	}
+	r.listener = listener
+
+	r.serverWaitGroup.Add(1)
+	go func() {
+		if err := r.server.Serve(listener); err != nil {
+			r.logger.WithError(err).Error("RPC server failed")
+		}
+		r.serverWaitGroup.Done()
+	}()
+
+	r.logger.Debugf("gRPC server is listening at %s", r.Endpoint())
+}
+
+// Endpoint returns RPC server address suitable for use in agent's "-api-endpoint" flag.
+func (r *RPC) Endpoint() string {
+	// Work around host.docker.internal missing on Linux
+	if runtime.GOOS == "linux" {
+		return r.listener.Addr().String()
+	}
+
+	port := r.listener.Addr().(*net.TCPAddr).Port
+
+	return fmt.Sprintf("host.docker.internal:%d", port)
+}
+
+// Stop gracefully stops the RPC server.
+func (r *RPC) Stop() {
+	r.server.GracefulStop()
+	r.serverWaitGroup.Wait()
+}
+
+func (r *RPC) InitialCommands(
+	ctx context.Context,
+	req *api.InitialCommandsRequest,
+) (*api.CommandsResponse, error) {
+	task, err := r.build.GetTaskFromIdentification(req.TaskIdentification, r.clientSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	return &api.CommandsResponse{
+		Environment:      r.build.Environment,
+		Commands:         task.ProtoTask.Commands,
+		ServerToken:      r.serverSecret,
+		TimeoutInSeconds: int64(task.Timeout.Seconds()),
+	}, nil
+}
+
+func getCommandAfter(commands []*api.Command, currentCommand string) string {
+	var doReturn bool
+
+	for _, command := range commands {
+		if doReturn {
+			return command.Name
+		}
+
+		if command.Name == currentCommand {
+			doReturn = true
+		}
+	}
+
+	return ""
+}
+
+func (r *RPC) ReportSingleCommand(
+	ctx context.Context,
+	req *api.ReportSingleCommandRequest,
+) (*api.ReportSingleCommandResponse, error) {
+	task, err := r.build.GetTaskFromIdentification(req.TaskIdentification, r.clientSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	var nextCommand string
+
+	logEntry := r.logger.WithFields(map[string]interface{}{
+		"task":    task.ProtoTask.Name,
+		"command": req.CommandName,
+	})
+
+	// Register whether the current command succeeded or failed
+	// so that the main loop can make the decision whether
+	// to proceed with the execution or not.
+	if req.Succeded {
+		logEntry.Debug("command succeeded")
```

Comment:
Let's add `CommandName` to the log message

Reply:
It's already in the log message, see the full change in the commit that introduces this (https://github.com/cirruslabs/cirrus-cli/pull/6/commits/1f89bcb54a9ab99f7f97b2c3d23fa7aec1ddb27d).